indexing-service/src/main/java/io/druid/indexing/common/task/HadoopTask.java

```diff
@@ -93,7 +93,7 @@ protected ClassLoader buildClassLoader(final TaskToolbox toolbox) throws Excepti
       driverURLs.addAll(Arrays.asList(((URLClassLoader) hadoopLoader).getURLs()));
     }
 
-    final URLClassLoader loader = new URLClassLoader(driverURLs.toArray(new URL[driverURLs.size()]), null);
+    final URLClassLoader loader = new URLClassLoader(driverURLs.toArray(new URL[driverURLs.size()]), Thread.currentThread().getContextClassLoader());
```

Comment:
hmmm, I started doing some testing with hadoop libs not on overlord classpath but only in the url-classloader and this change breaks things. It appears putting the thread classloader in there mixes up things where some classes being used are from thread context classloader and that is leading to ClassNotFoundException for hadoop classes.